Common/labels.cpp

```diff
@@ -138,10 +138,13 @@ bool Labels::syncInts(const std::set<int> &values)
 	std::set<int>	valuesToAdd		= values,
 					valuesToRemove;
 
-	for (const Label & label : _labels)
+	for (Label & label : _labels)
 	{
 		int value = label.value();
 
+		if (!label.hasIntValue())
+			label.setValue(value);
```

Comment:
The function `bool syncInts` should tell if it changed right?

So what could happen, possibly, is that all the values were maybe all already convertible to the ints that are being synced. Then you do `setValue` which changes it so that it "hasIntValue", Im not sure if that should count as a "change" but I think it wouldn't hurt to do so.

So let the function remember if it did `setValue` on at least one of the labels and `return  setValueAtLeastOnce || valuesToAdd.size() + valuesToRemove.size() > 0;` or something like that.

Reply:
OK I have added your remark.